charts/gitops-runtime/templates/hooks/pre-install/validate-values.yaml

```diff
@@ -31,15 +31,93 @@ spec:
         image: "{{ .Values.installer.image.repository }}:{{ .Values.installer.image.tag | default .Chart.Version }}"
         imagePullPolicy: {{ .Values.installer.image.pullPolicy }}
         env:
+        - name: ARGOCD_LABELS
+          value: "{{ range $k, $v := .Values.installer.argoCdVersionCheck.argoServerLabels }}{{ $k }}={{ $v }},{{ end }}"
+        - name: ARGOCD_VERSION_PATH
+          value: {{ .Values.installer.argoCdVersionCheck.versionPath | default "/api/version" }}
+        - name: ARGOCD_SCHEME
+          value: {{ .Values.installer.argoCdVersionCheck.scheme | default "http" }}
+        - name: ARGOCD_INSECURE
+          value: {{ .Values.installer.argoCdVersionCheck.insecureSkipVerify | default false | quote }}
+        - name: REQUIRED_VERSION_CONSTRAINT
+          value: ">=2.12 <3"
         - name: NAMESPACE
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
-        - name: VERSION
+        - name: CHART_VERSION
           value: {{ .Chart.Version }}
         command: ["sh", "-c"]
-        args: 
-        - cf helm validate --values /job_tmp/values.yaml --namespace ${NAMESPACE} --version ${VERSION} --hook --log-level debug
+        args:
+        - |
```

Comment:
If you use VSCode I advise using [YAML Embedded Languages](https://marketplace.visualstudio.com/items/?itemName=harrydowning.yaml-embedded-languages) languages extension and adding language comment - makes it easier on eyes.
![image](https://github.com/user-attachments/assets/a1255465-b548-4b35-a9b6-8de53ef88cc3)

Reply:
that's a nice trick. i installed the extension, and added the comment.

Reply:
You can also run this code like so: `cat argo-platform/merge-queue.yaml | yq '.steps.wait_for_previous_promotion.commands[0]' | bash`. Although figuring out variables passed to the script is tricky.